code/game/objects/items/storage/storage.dm

```diff
@@ -721,24 +720,25 @@ W is always an item. stop_warning prevents messaging. user may be null.**/
 			to_chat(user, SPAN_WARNING("[ammo_dumping] is empty."))
 	return TRUE
 
-/obj/item/storage/proc/dump_into(obj/item/storage/M, mob/user)
+/obj/item/storage/proc/dump_into(obj/item/storage/Main, mob/user)
+
 	if(user.action_busy)
 		return
 
-	if(!M.contents.len)
-		to_chat(user, SPAN_WARNING("[M] is empty."))
+	if(!Main.contents.len)
+		to_chat(user, SPAN_WARNING("[Main] is empty."))
 		return
-	if(!has_room(M.contents[1])) //Does it have room for the first item to be inserted?
+	if(!has_room(Main.contents[1])) //Does it have room for the first item to be inserted?
 		to_chat(user, SPAN_WARNING("[src] is full."))
 		return
 
-	to_chat(user, SPAN_NOTICE("You start refilling [src] with [M]."))
+	to_chat(user, SPAN_NOTICE("You start refilling [src] with [Main]."))
 	if(!do_after(user, 1.5 SECONDS, INTERRUPT_ALL, BUSY_ICON_GENERIC))
 		return
-	for(var/obj/item/I in M)
+	for(var/obj/item/I in Main)
 		if(!has_room(I))
```

Comment:
Currently this will return TRUE even though your handle_item_insertion will return false because that is the one where you put the logic to keep a space for the flaregun.

So effectively right now if you fill a flaregun belt without a flaregun holstered, the extra flares spill out onto the ground. Sort of funny, but I doubt you intend this.

https://github.com/cmss13-devs/cmss13/assets/76988376/be9b8b79-bb78-49ea-8d22-c349508cae32

So:
- Either you should be overriding the logic for has_room for a `/obj/item/storage/belt/gun/flaregun` (possibly instead of handle_item_insertion)
- Or you also put the handle_item_insertion call into the if to break (e.g. `if(!has_room(new_item) || !(handle_item_insertion(new_item, TRUE, user))`
- Or just place handle_item_insertion into its own if, put the flare back into the pouch, and then break the loop.

Be sure to test your changes and think of possible edge cases like I did here.

Reply:
i tested it with or without the flare gun holster and i didn't saw it ... weird
i had this happening in previous version when i tested it so i am not surprise....
unsure what to do i will look into it.

Reply:
i try to reproduce it after fixing everything else you commented
and i can't reproduce what you show on the video all i get is a the rig is full.....

Reply:
oh i did reproduce it when i hold the belt in my hand.... weird...

Reply:
It will correctly think its full if its actually full. But if its not actually full (because it has an empty space for the gun) it will dump extras on the floor. My video has it on the belt slot in hand - but shouldn't matter.

Reply:
your right i manage to reproduce it after my third just didn't update here should work tested it with part empty and nothing fall on the floor...